File: Python/Cpu.py
from Komponen import Komponen
from Gpu import Gpu

class Cpu(Komponen):
    def __init__(self, jumlah_core=0, kecepatan_ghz=0.0, merk="", nama="", gpu=None):
        super().__init__(merk, nama)
        self.jumlah_core = jumlah_core
        self.kecepatan_ghz = kecepatan_ghz
        self.gpu = gpu if gpu else Gpu()

    def set_jumlah_core(self, jumlah_core):
        self.jumlah_core = jumlah_core
    
    def set_kecepatan_ghz(self, kecepatan_ghz):
        self.kecepatan_ghz = kecepatan_ghz
    
    def set_gpu(self, gpu):
        self.gpu = gpu
    
    def get_jumlah_core(self):
        return self.jumlah_core
    
    def get_kecepatan_ghz(self):
        return self.kecepatan_ghz
    
    def get_gpu(self):
        return self.gpu

File: CPP/Cpu.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"
#include "Gpu.cpp"


using namespace std;


class Cpu : public Komponen
{
private:
    int jumlahCore;
    float kecepatanGHz;
    Gpu gpu;
public:
    Cpu()
    {


    }


    Cpu(int jumlahCore, float kecepatanGHz, string merk, string nama, Gpu gpu) : Komponen(merk, nama)
    {
        this->jumlahCore = jumlahCore;
        this->kecepatanGHz = kecepatanGHz;
        this->gpu = gpu;
    }


    void setJumlahCore(int jumlahCore)
    {
        this->jumlahCore = jumlahCore;
    }


    void setKecepatanGHz(float kecepatanGHz)
    {
        this->kecepatanGHz = kecepatanGHz;
    }

    void setGpu(Gpu gpu)
    {
        this->gpu = gpu;
    }

    
    int getJumlahCore()
    {
        return this->jumlahCore;
    }
    
    
    float getKecepatanGHz()
    {
        return this->kecepatanGHz;
    }
    
    Gpu getGpu()
    {
        return this->gpu;
    }

    ~Cpu()
    {


    }
};
